packages/cli/src/utils/codegen.ts

```diff
@@ -20,11 +22,11 @@ export async function generateLibDeploy(configPath: string, out: string, systems
   console.log("Generating deployment script");
   const LibDeploy = await ejs.renderFile(path.join(contractsDir, "LibDeploy.ejs"), config, { async: true });
   const libDeployPath = path.join(out, "LibDeploy.sol");
-  writeFileSync(libDeployPath, LibDeploy);
+  await writeFile(libDeployPath, LibDeploy);
 
   return libDeployPath;
 }
 
 export async function resetLibDeploy(out: string) {
-  writeFileSync(path.join(out, "LibDeploy.sol"), readFileSync(path.join(contractsDir, "LibDeploy.sol")));
+  await writeFile(path.join(out, "LibDeploy.sol"), await originalLibDeploy);
```

Comment:
I'm a bit wary of caching the original LibDeploy and restoring it here - it might lead folks to believe they can change the generated LibDeploy.sol manually and then confuse them about why their changes are not actually picked up. For this reason I'd prefer restoring the "stub version" of LibDeploy (https://github.com/latticexyz/mud/blob/main/packages/cli/src/contracts/LibDeploy.sol) after running the script, which should also make keep imports/hot reload working, but include a fat warning about it being a stub

Reply:
> For this reason I'd prefer restoring the "stub version" of LibDeploy (https://github.com/latticexyz/mud/blob/main/packages/cli/src/contracts/LibDeploy.sol) 

This _is_ what it's doing.

Maybe you could spin this up locally to see what I am seeing? I started with the mud-template-minimal, linked mud deps locally (including CLI package), and when I went to make any changes to contracts, `LibDeploy.sol` would get overwritten or deleted, thus breaking hot reloading for any future changes.

Reply:
Ohh I see, `originalLibDeploy` actually refers to the stub in the cli package. When reading this yesterday I thought it used the `LibDeploy.sol` file in the user directory and restoring it after deploying.

Could we rename the variable to `stubLibDeploy` or something to make it obvious even to devs reading this late at night? 😄 (This was already unclear before your change, good call to move reading the file outside of the reset function!)